management-interface.go

```diff
@@ -215,6 +234,74 @@ func CheckInterfaceHandler(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(response)
 }
 
+// SpeakerTestHandler will show a frame from the camera to help with positioning
+func SpeakerTestHandler(w http.ResponseWriter, r *http.Request) {
+	tmpl.ExecuteTemplate(w, "speaker-test.html", nil)
+}
+
+// fileExists returns whether the given file or directory exists
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+// findAudioFile locates our test audio file.  It returns true and the location of the file
+// if the file is found. And false and empty string otherwise.
+func findAudioFile(fileName string, secondaryPath string) (bool, string) {
+
+	// Check if the file is in the executable directory
+	if executablePath != "" && fileExists(executablePath+fileName) {
+		return true, executablePath
+	}
+
+	// In our default, second location?
+	if fileExists(secondaryPath + fileName) {
+		return true, secondaryPath
+	}
+
+	// Test sound not available
+	log.Printf("File " + fileName + " NOT found")
+	return false, ""
+
+}
+
+// SpeakerStatusHandler attempts to play a sound on connected speaker(s).
+func SpeakerStatusHandler(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	response := make(map[string]string)
+
+	fileName := "/IEEE_float_mono_32kHz.wav"         // Default sound file name.
+	secondaryPath := "/usr/lib/management-interface" // Check here if the file is not found in the executable directory.
+
+	result, testAudioPath := findAudioFile(fileName, secondaryPath)
+	if result {
+		// Play the sound file
+		args := []string{"-v50", "-q", testAudioPath + fileName}
+		output, err := exec.Command("play", args...).CombinedOutput()
+		w.WriteHeader(http.StatusOK)
```

Comment:
If the file isn't found or the `play` command fails, a `http.StatusInternalServerError` should be returned. You then don't need the `status` field.